### PR TITLE
Fix build of sunxi-meminfo and sunxi-script_extractor and possible future programs

### PR DESCRIPTION
**Pull request by daym**
See issue #70

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -111,7 +111,7 @@ sunxi-nand-part: nand-part-main.c nand-part.c nand-part-a10.h nand-part-a20.h
 	$(CC) $(LDFLAGS) -o $@ nand-part-main.o nand-part-a10.o nand-part-a20.o $(LIBS)
 
 sunxi-%: %.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$<) $(LIBS)
```
Comment (n1tehawk) [inline Makefile]: Keeping `$filter()` but applying it to `$<` now doesn't really make sense, or? As it is, this change breaks compilation of `sunxi-fexc`.

File: Makefile [hunk 2]
```diff
@@ -153,10 +153,10 @@ boot_head_sun5i.elf: boot_head.S boot_head.lds
 sunxi-bootinfo: bootinfo.c
 
 sunxi-meminfo: meminfo.c
-	$(CROSS_COMPILE)gcc -g -O0 -Wall -static -o $@ $^
+	$(CROSS_COMPILE)gcc -g -O0 -Wall -static -o $@ $<
 
 sunxi-script_extractor: script_extractor.c
-	$(CROSS_COMPILE)gcc -g -O0 -Wall -static -o $@ $^
+	$(CROSS_COMPILE)gcc -g -O0 -Wall -static -o $@ $<
```
Comment (n1tehawk) [inline Makefile]: My impression is that we're missing an actual cross-compile rule for `sunxi-pio` here. I would favor introducing some common *TARGET_CFLAGS* and then use something like:
```Makefile
# target tools
TARGET_CFLAGS = -std=c99 $(DEFINES) -Iinclude/ -g -O0 -Wall -static
sunxi-meminfo: meminfo.c
	$(CROSS_COMPILE)gcc $(TARGET_CFLAGS) -o $@ $<
sunxi-pio: pio.c
	$(CROSS_COMPILE)gcc $(TARGET_CFLAGS) -o $@ $<
sunxi-script_extractor: script_extractor.c
	$(CROSS_COMPILE)gcc $(TARGET_CFLAGS) -o $@ $<
```

File: Makefile (hunk 1 above)
Comment (daym) [inline Makefile]: Yeah, sorry, it's too paranoid now. 

I don't really like these magical rules - I think it's much preferrable and less error-prone to have an explicit rule for each target file.

File: Makefile (hunk 2 above)
Comment (daym) [inline Makefile]: I agree.

File: Makefile [hunk 3]
```diff
@@ -102,7 +102,7 @@ LIBUSB_CFLAGS = `pkg-config --cflags $(LIBUSB)`
 LIBUSB_LIBS = `pkg-config --libs $(LIBUSB)`
 
 sunxi-fel: fel.c fel-to-spl-thunk.h progress.c progress.h
-	$(CC) $(CFLAGS) $(LIBUSB_CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^) $(LIBS) $(LIBUSB_LIBS)
+	$(CC) $(CFLAGS) $(LIBUSB_CFLAGS) $(LDFLAGS) -o $@ fel.c progress.c $(LIBS) $(LIBUSB_LIBS)
```
Comment (daym) [inline Makefile]: You are right, it should be fine to use $(filter ) in this case - as it was in the beginning.

**Comment by n1tehawk** [comment]
The PR was obsoleted by merging #73

**Pull request closed without merging**